File: Day 3/day3.py
def organize_inventory(inventory: dict) -> dict:
  invent_org = {}
  for invent in inventory:
    name, quan, cat = invent['name'], invent['quantity'], invent['category']
    invent_org.setdefault(cat, {name:0})
    invent_org[cat][name] = invent_org[cat].get(name, 0) + quan

  return invent_org

File: Day3/day3.ts
function organizeInventory(inventory: {name: string, quantity:number, category:string}[]): Record<string, Record<string,number>> {
    const inventOrg: Record<string, Record<string,number>> = {};


    for (const invent of inventory){
        const {name, quantity, category } = invent;

        if (!inventOrg[category]){ inventOrg[category] = {};}

        inventOrg[category][name] = (inventOrg[category][name] || 0) + quantity;
    }

    return inventOrg;
}